diff --git a/src/fauxmoESP.cpp b/src/fauxmoESP.cpp
index 71a975f..2abd641 100644
--- a/src/fauxmoESP.cpp
+++ b/src/fauxmoESP.cpp
@@ -37,8 +37,8 @@ void fauxmoESP::_sendUDPResponse() {
 
 	DEBUG_MSG_FAUXMO("[FAUXMO] Responding to M-SEARCH request\n");
 
-	IPAddress ip = WiFi.localIP();
-    String mac = WiFi.macAddress();
+    IPAddress ip = _ip_set ? _ip : WiFi.localIP();
+    String mac = _mac_set ? _mac : WiFi.macAddress();
     mac.replace(":", "");
     mac.toLowerCase();
 
@@ -178,8 +178,8 @@ bool fauxmoESP::_onTCPDescription(AsyncClient *client, String url, String body)
 
 	DEBUG_MSG_FAUXMO("[FAUXMO] Handling /description.xml request\n");
 
-	IPAddress ip = WiFi.localIP();
-    String mac = WiFi.macAddress();
+    IPAddress ip = _ip_set ? _ip : WiFi.localIP();
+    String mac = _mac_set ? _mac : WiFi.macAddress();
     mac.replace(":", "");
     mac.toLowerCase();
 
@@ -650,7 +650,8 @@ void fauxmoESP::enable(bool enable) {
 		#ifdef ESP32
             _udp.beginMulticast(FAUXMO_UDP_MULTICAST_IP, FAUXMO_UDP_MULTICAST_PORT);
         #else
-            _udp.beginMulticast(WiFi.localIP(), FAUXMO_UDP_MULTICAST_IP, FAUXMO_UDP_MULTICAST_PORT);
+            IPAddress ip = _ip_set ? _ip : WiFi.localIP();
+            _udp.beginMulticast(ip, FAUXMO_UDP_MULTICAST_IP, FAUXMO_UDP_MULTICAST_PORT);
         #endif
         DEBUG_MSG_FAUXMO("[FAUXMO] UDP server started\n");
 
diff --git a/src/fauxmoESP.h b/src/fauxmoESP.h
index facb13e..2e55a0c 100644
--- a/src/fauxmoESP.h
+++ b/src/fauxmoESP.h
@@ -109,6 +109,8 @@ class fauxmoESP {
         void enable(bool enable);
         void createServer(bool internal) { _internal = internal; }
         void setPort(unsigned long tcp_port) { _tcp_port = tcp_port; }
+        void setIP(IPAddress ip) { _ip = ip; _ip_set = true; }
+        void setMac(const char * mac) { _mac = String(mac); _mac.replace(":", ""); _mac.toLowerCase(); _mac_set = true; }
         void handle();
 
     private:
@@ -117,6 +119,10 @@ class fauxmoESP {
         bool _enabled = false;
         bool _internal = true;
         unsigned int _tcp_port = FAUXMO_TCP_PORT;
+        IPAddress _ip;
+        String _mac;
+        bool _ip_set = false;
+        bool _mac_set = false;
         std::vector<fauxmoesp_device_t> _devices;
 		#ifdef ESP8266
         WiFiEventHandler _handler;
